pandas/indexes/category.py

```diff
@@ -136,6 +136,19 @@ def _simple_new(cls, values, name=None, categories=None, ordered=None,
         result._reset_identity()
         return result
 
+    @Appender(_index_shared_docs['_shallow_copy'])
+    def _shallow_copy(self, values=None, categories=None, ordered=None,
+                      **kwargs):
```

Comment:
I don't see why this is necessary, the ``_simple_new`` already handles this, if ``categories`` or ``ordered`` is ``not None`` they will be set, otherwise they won't and will simply be a refernce to the data, which is a ``CategoricalIndex`` at this point.